src/source/tile.js

```diff
@@ -148,13 +146,20 @@ class Tile {
             return;
         }
 
-        if (data.rawTileData) {
-            // Only vector tiles have rawTileData
-            this.rawTileData = data.rawTileData;
+        if (data.featureIndex) {
+            this.latestFeatureIndex = data.featureIndex;
+            if (data.rawTileData) {
+                // Only vector tiles have rawTileData, and they won't update it for
+                // 'reloadTile'
+                this.latestRawTileData = data.rawTileData;
+                this.latestFeatureIndex.rawTileData = data.rawTileData;
+            } else if (this.latestRawTileData) {
+                // If rawTileData hasn't updated, hold onto a pointer to the last
+                // one we received
+                this.latestFeatureIndex.rawTileData = this.latestRawTileData;
```

Comment:
In what cases does a the new `data` have a `featureIndex` but not a `rawTileData`?

Reply:
`loadTile` in `VectorTileWorkerSource` returns a `rawTileData`, but `reloadTile` doesn't (we use a new `FeatureIndex` but with the previous data since it hasn't changed).

Reply:
Thanks!